runtime/service.go

```diff
@@ -1183,42 +1185,55 @@ func (s *service) shutdown(
 	shutdownCtx, cancel := context.WithTimeout(requestCtx, timeout)
 	defer cancel()
 	go func() {
-		// Once the shutdown procedure is done, the shim needs to shutdown too.
-		// This also ensures that if the VMM is still alive, it will receive a
-		// SIGKILL via exec.CommandContext
-		<-shutdownCtx.Done()
+		// Once the firecracker process is complete, the shim needs to shutdown too.
+		s.machine.Wait(context.Background())
```

Comment:
Do we want to log error here just in case?

Reply:
It may double-log the error, but since err would be nil most of the case, I think it is better to double-log rather than ignoring. So will do.